### ISSUE TITLE
False positive PropertyNotSetInConstructor and possible false positive MismatchingDocblockParamType

### ISSUE THREAD

**Issue by ErikBooijCB** [vimeo/psalm#792]
Example:

https://getpsalm.org/r/2d4145490c

```php
<?php

interface PercolatorQueryOptimizerInterface {}
  
class PercolatorQueryBuilder
{
    /** @var PercolatorQueryOptimizerInterface[] */
    private $queryOptimizers;

    /**
     * @param PercolatorQueryOptimizerInterface[] $queryOptimizers
     */
    public function __construct(PercolatorQueryOptimizerInterface ...$queryOptimizers)
    {
        $this->queryOptimizers = $queryOptimizers;
    }
}
```

It says the property `$queryOptimizers` isn't set in the constructor. Also it indicates the docblock is a mismatch, but it actually seems valid to me.

**Comment by muglug** [comment]
Yeah, the false positive here is the mismatching docblock issue, which somehow then prevents further analysis. Making the docblock more precise makes Psalm happy: either https://getpsalm.org/r/fc89bef029 or https://getpsalm.org/r/56b0425583 will do

**Comment by ErikBooijCB** [comment]
Ah, sweet we can work from there, I can live with a `@psalm-suppress` until a future release. Thanks for the help!
